src/main/java/com/commercetools/sync/inventory/InventorySync.java

```diff
@@ -0,0 +1,380 @@
+        InventorySyncOptions> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InventorySync.class);
+
+    /*
+     * Holds threads for executing sync process in parallel. Should be instantiated when parallel processing is
+     * enabled via InventorySyncOptions.
+     */
+    private ExecutorService executorService = null;
+
+    //Cache that maps supply channel key to supply channel Id for supply channels existing in database.
+    private Map<String, String> supplyChannelKeyToId;
+
+    private InventoryService inventoryService;
+
+    private TypeService typeService;
+
+    public InventorySync(@Nonnull final InventorySyncOptions syncOptions) {
+        this(syncOptions, new InventoryServiceImpl(syncOptions.getCtpClient().getClient()),
+                new TypeServiceImpl(syncOptions.getCtpClient().getClient()));
+    }
+
+    InventorySync(final InventorySyncOptions syncOptions, final InventoryService inventoryService,
+                  final TypeService typeService) {
+        super(new InventorySyncStatistics(), syncOptions, LOGGER);
+        this.inventoryService = inventoryService;
+        this.typeService = typeService;
+        if (syncOptions.getParallelProcessing() > 1) {
+            executorService = Executors.newFixedThreadPool(syncOptions.getParallelProcessing());
+        }
+    }
+
+    /**
+     * <p>Performs full process of synchronisation between inventory entries present in a system
+     * and passed {@code inventories}. This is accomplished by:
+     * <ul>
+     *     <li>Comparing entries and drafts by {@code sku} and {@code supplyChannel} key</li>
+     *     <li>Calculating of necessary updates and creation commands</li>
+     *     <li>Actually <strong>performing</strong> changes in a target CTP project by sphere calls</li>
+     * </ul>
+     * The process is customized according to {@link InventorySyncOptions} passed to constructor of this object.
+     * After the process finishes you can obtain its summary by calling {@link InventorySync#getStatistics()}.</p>
+     * <p><strong>This method is meant to be called once. For new sync process please create new {@link InventorySync}
+     * instance!</strong></p>
+     *
+     * @param inventories {@link List} of {@link InventoryEntryDraft} containing data that would be synced into
+     *                                CTP project.
+     */
+    @Override
+    protected void processDrafts(@Nonnull List<InventoryEntryDraft> inventories) {
+        buildChannelMap();
+        List<InventoryEntryDraft> accumulator = new LinkedList<>();
+        for (InventoryEntryDraft entry : inventories) {
+            if (entry != null) {
+                if (entry.getSku() != null) {
+                    accumulator.add(entry);
+                    if (accumulator.size() == syncOptions.getBatchSize()) {
+                        runDraftsProcessing(accumulator);
+                        accumulator = new LinkedList<>();
```

Comment:
And actually you can create ArrayList with initial size to avoid resizing since you know the size is `syncOptions.getBatchSize()`.